TelCo.ColorCoder/ColorMap.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Drawing;
+
+namespace TelCo.ColorCoder
+{
+    /// <summary>
+    /// This class provides the color coding and mapping of pair number to color and color to pair number.
+    /// </summary>
+    class ColorMap
+    {
+        #region Private Variables
+        /// <summary>
+        /// Array of Major and Minor colors
+        /// </summary>
+        private static Color[] major, minor;
+        #endregion
+
+        #region Constructor
+        static ColorMap() {
+            major = new Color[] { Color.White, Color.Red, Color.Black, Color.Yellow, Color.Violet };
+            minor = new Color[] { Color.Blue, Color.Orange, Color.Green, Color.Brown, Color.SlateGray };
+        }
+        #endregion
+
+        /// <summary>
+        /// data type defined to hold the two colors of clor pair
+        /// </summary>
+        internal class ColorPair
+        {
+            internal Color majorColor, minorColor;
+            public override string ToString() {
+                return string.Format("MajorColor:{0}, MinorColor:{1}", majorColor.Name, minorColor.Name);
+            }
+        }
+
+        #region Methods
+        /// <summary>
+        /// Given a pair number function returns the major and minor colors in that order
+        /// </summary>
+        /// <param name="pairNumber">Pair number of the color to be fetched</param>
+        /// <returns></returns>
+        public ColorPair GetColorFromPairNumber(int pairNumber) {
+            // The function supports only 1 based index. Pair numbers valid are from 1 to 25
+            if (pairNumber < 1 || pairNumber > (minor.Length) * (major.Length))
+                throw new ArgumentOutOfRangeException(string.Format("Argument PairNumber:{0} is outside the allowed range", pairNumber));
+            
+            // Find index of major and minor color from pair number
+            int zeroBasedPairNumber = pairNumber - 1, majorIndex = zeroBasedPairNumber / minor.Length, minorIndex = zeroBasedPairNumber % minor.Length;
+            return new ColorPair() { majorColor = major[majorIndex], minorColor = minor[minorIndex] };
+        }
+        
+        /// <summary>
+        /// Given the two colors the function returns the pair number corresponding to them
+        /// </summary>
+        /// <param name="pair">Color pair with major and minor color</param>
+        /// <returns></returns>
+        public int GetPairNumberFromColor(ColorPair pair) {
```

Comment:
Why don't we separate finding Minor and Major color logic ?

Reply:
Created new functions for finding the major and minor color index. 
Hence each function will have their own single functionality/responsibility.